augur/frequency_estimators.py

```diff
@@ -1,14 +1,17 @@
 # estimates clade frequencies
 from __future__ import division, print_function
-from collections import defaultdict
+from collections import defaultdict, deque
 import datetime
+from dateutil.relativedelta import relativedelta
```

Comment:
If the sole reason for this new dependency is to calculate relative dates, that can be achieved by using the existing dependency `isodate` (added for a similar reason in #740). See this alternative: 2e571e1412ed578a0aa767e467b6b521eaf81b6e

Reply:
Good call! I missed this in my list above. In addition to avoiding an extra explicit dependency, the isodate library is closer to the interface we might have used if we thought about the pivot interval units more holistically at the time we added them. We added that interface in January 2021 and didn't add relative dates for augur filter until June 2021, after much more thought going into the ISO-8601 interface. Using isodate in pivot calculations paves the way for support of any interval length and unit supported by the ISO-8601 standard.

Reply:
I just merged 2e571e1.